enhancements/microshift/microshift-router-configuration.md

```diff
@@ -0,0 +1,466 @@
+    interfaces:
+    - <NIC name>
+    ipAddresses:
+    - <IP address>
+```
+
+### Enable/disable the router
+The following configuration is proposed:
+```yaml
+ingress:
+  status: <Enabled|Disabled> # Defaults to Enabled.
+```
+
+With this option MicroShift will decide whether it should create the router
+upon starting. This includes not just the pod, but all the associated resources
+that come along with it: namespace, services, configmaps, etc.
+
+When setting the option to `Disabled`, the next MicroShift restart should delete
+all the router related resources, including APIs and controllers.
+
+Setting the option to `Enabled` will deploy the router, while `Disabled` will disable
+it. Default will be `Enabled`.
+
+### Listening ports
+The following configuration is proposed:
+```yaml
+ingress:
+  ports:
+    http: <int> # Defaults to 80.
+    https: <int> # Defaults to 443.
+```
+
+MicroShift does not own the host, which means there might be other
+applications running alongside and some ports may already be opened.
+MicroShift must be able to accommodate such situations and be ready to listen
+in other ports, hence the configuration options.
+
+In order to allow this configuration and many other advantages, the
+router service shall be changed so that it is exposed using `LoadBalancer`
+type instead of using host ports. See the Design Details section for more
+information.
+
+#### Firewalling ports
+Using `LoadBalancer` service type prevents the usage of firewalld to block
+access to ports. Since this is a feature that users may require for auditing,
+an alternative is provided.
+
+The `NetworkPolicy` resource fits the needs for blocking traffic to services
+and pods. Policies can also be configured to allow specific IP addresses and
+provide a high degree of customization for network traffic. They use the CNI
+instead of host level settings, providing greater flexibility and being self
+contained within MicroShift.
+
+Users in need of blocking traffic shall use the `NetworkPolicy` resource. A
+section in the documentation will be added to describe how to do this.
+
+### Configure IPs where the router listens
+The following configuration is proposed:
+```yaml
+ingress:
+  expose: # Defaults to IPs in the host. Details below.
+    hostnames:
+    - <hostname>
+    interfaces:
+    - <NIC name>
+    ipAddresses:
+    - <IP address>
+```
+As we have seen in previous sections, the use of `LoadBalancer` makes ovnk
+configure iptables rules to expose the service outside of the cluster.
+Building on [LoadBalancer controller enhancement](loadbalancer-service-support.md),
+the service is exposed using the node IP and configured ports.
+
+All the IPs where the router is exposed need to be part of the `LoadBalancer`
+service status definition. In order to provide more flexibility, the
+configuration options are able to take IP addresses, hostnames and/or NIC
+names.
+
+MicroShift will default to listen to external IPs, the service IP and the
```

Comment:
What will happen when `expose` is configured? Will it override the default behavior (all IPs: ext, svc, and kas) completely or it will be an addition to the defaults?

Reply:
yes, it overrides all the defaults and only listens where it was configured to do so. Added a clarification for that.